### ISSUE TITLE
protect assets still have hash in the file link, and the old file link doesn't redirect to the new file 

### ISSUE THREAD

**Issue by normann** [silverstripe/silverstripe-framework#9142]
## Affected Version

Replicated in 4.4.1, might be replicable in 4.4.0

## Description

Give the fact that version 4.4 is trying to solve the file link dynamically changing due to a hash (to refect file version) in the URL, once the file is replaced by another file where we can see the issue prior to version 4.3 (inclusive), but the fixes or the feature seemed not affecting protected assets

## Steps to Reproduce

1. [OK] Login as Admin and visit 'admin/assets'
2. [OK] Create a Folder and set the folder to be protected, i.e. config those **Who can view** to _Only These People_ and **Viewer Group** to an _Administration_ group
3. [OK] Upload a file to the folder created in step 2, edit the file and **Save and Publish**
4. [OK] Insert a link pointing to the file created in step 3 to a page, *Save and Publish** the Page
5. [Fail] Landing the page in another tab, copy the file's address and paste it to a text editor, observe that the link, expecting it is not containing the versioning has in the URL, but it still containing a hash
6. [OK] Paste the file's URL in a new tab of the same browser, the file is accessible.
7. [OK] Go to the file's editing interface, make a replacement operation on the file, with a new file (either the same file with some changes or another file so we could easily compare later)
8. [OK] Refresh the page with the file link, inspect the file link has changed with a new hash in the URL.
9. [OK] Click on the file link or open it in a new tab, the content accessed shows it is the new file.
10 [Fail] Revisit the old File link that is copied and put it in the text editor in Step 5, expecting it is either show the new's File's content or redirecting to the new file link. But, the link simply shows a "Not Found"

**Comment by normann** [comment]
If Step 5 is not considered as a failure, i.e. this is supposed behavior by SS4.4, at least Step 10 should be considered as a failure, right? An old URL bookmarked by site user should either redirect to the new URL or load with the new file content.

**Comment by michalkleiner** [comment]
The issue might be better logged on the https://github.com/silverstripe/silverstripe-asset-admin repo.

**Comment by normann** [comment]
Yes, noticed that after the creation of the issue :) @michalkleiner 
I think I will manage to migrate the issue to either `SilverStripe-asset-admin` or `SilverStripe-assets `(now they are separated), will ask SilverStripe though.

**Comment by ScopeyNZ** [comment]
I'd say this probably belongs in assets, not asset-admin. The administration screens are being used correctly but the URLs generated to serve the assets are wrong - which implies it's `assets`. I'll move it for you.
